.github/workflows/docker.yml

```diff
@@ -0,0 +1,276 @@
+# Copyright 2021 Security Scorecard Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: docker-build
+on: [push, pull_request]
+jobs:
+  scorecard:
+    name: scorecard-docker
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+     - name: Install Protoc
+       uses: arduino/setup-protoc@64c0c85d18e984422218383b81c52f8b077404d3 # v1.1.2
+       with:
+        version: 3.17.3
+     - name: Cache builds
+       # https://github.com/mvdan/github-actions-golang#how-do-i-set-up-caching-between-builds
+       uses: actions/cache@937d24475381cd9c75ae6db12cb4e79714b926ed #v2.1.7
+       with:
+          # In order:
+          # * Module download cache
+          # * Build cache (Linux)
+          # * Build cache (Mac)
+          # * Build cache (Windows)
+         path: |
+           ~/go/pkg/mod
+           ~/.cache/go-build
+           ~/Library/Caches/go-build
+           %LocalAppData%\go-build
+         key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+         restore-keys: |
+           ${{ runner.os }}-go-
+     - name: Clone the code
+       uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.3.4
+       with:
+          fetch-depth: 0
+     - name: Setup Go
+       uses: actions/setup-go@bfdd3570ce990073878bf10f6b2d79082de49492 # v2.2.0
+       with:
+         go-version: '^1.17'
+     - name: docker build
+       run: make scorecard-docker
```

Comment:
Could we parallelize this by saying `make -j X dockerbuild`? That way, a new docker build steps needs to be added only to the Makefile and not in 2 places (Makefile and here).

Reply:
Let me know if you disagree.